File: 21558 - Nithin/week 2/901-online-stock-span/901-online-stock-span.cpp
class StockSpanner {
public:
    stack<pair<int, int>> stck;
    StockSpanner() {
        
    }
    
    int next(int price) {
        int stprice_span = 1;
        while(stck.empty() == false && price>=stck.top().first){
            stprice_span += stck.top().second;
            stck.pop();
        }
        stck.push({price, stprice_span});
        return stprice_span;
        
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */

File: 21561 - Suvam Tamang/Week 2/901_Online Stock Span.java
class StockSpanner {
    Stack<int []> tupleStack;
   
    public StockSpanner() {
        tupleStack = new Stack<>();
    }
    
    public int next(int price) {
        int result = 1;
        int arr [] = new int[2];
        
        while(!tupleStack.empty() && tupleStack.peek()[0]<=price){
            result += tupleStack.pop()[1];
        }
        
        arr[0]=price;
        arr[1]=result;
        tupleStack.push(arr);
        return result;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */
